internal/versions/versions_test.go

```diff
@@ -214,6 +252,71 @@ var _ = Describe("list versions", func() {
 			Expect(h.IsOpenshiftVersionSupported("unknown")).Should(BeFalse())
 		})
 	})
+
+	Context("AddOpenshiftVersion", func() {
+		var (
+			pullSecret   = "test_pull_secret"
+			releaseImage = "releaseImage"
+			ocpVersion   = "4.7.0"
+			keyVersion   = "4.7"
+		)
+
+		It("added version successfully", func() {
+			h := NewHandler(logger, mockRelease, versions, customOpenShiftVersions, "")
+			mockRelease.EXPECT().GetOpenshiftVersion(
+				gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(ocpVersion, nil).AnyTimes()
+
+			version, err := h.AddOpenshiftVersion(releaseImage, pullSecret)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			versionKey, err := h.GetSupportedKeyVersionFormat(ocpVersion)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			versionFromCache := h.openshiftVersions[versionKey]
+			Expect(*version.DisplayName).Should(Equal(ocpVersion))
+			Expect(h.GetReleaseVersion(keyVersion)).Should(Equal(ocpVersion))
+			Expect(h.GetReleaseImage(keyVersion)).Should(Equal(releaseImage))
+			Expect(h.GetRHCOSImage(keyVersion)).Should(Equal(*versionFromCache.RhcosImage))
+			Expect(h.GetRHCOSVersion(keyVersion)).Should(Equal(*versionFromCache.RhcosVersion))
+		})
+
```

Comment:
Add a scenario to check `supportLevel` when it is used from cache and when it's not.